packages/fuels/test/features/dev-2.test.ts

```diff
@@ -0,0 +1,79 @@
+import { execSync, spawn } from 'child_process';
+import { mkdirSync, rmSync } from 'fs';
+import { tmpdir } from 'os';
+import path from 'path';
+
+import { deferPromise, randomUUID } from '../../src';
+import { findChildProcessPid } from '../utils/findChildProcessPid';
+import { isProcessRunning } from '../utils/isProcessRunning';
+
+function runInit() {
+  const fuelsPath = path.join(process.cwd(), 'packages/fuels');
+
+  const init = path.join(tmpdir(), '.fuels', 'tests', randomUUID());
+
+  mkdirSync(init, { recursive: true });
+
+  execSync('pnpm init', { cwd: init });
+  execSync(`pnpm link ${fuelsPath}`, { cwd: init });
+
+  const contractDir = path.join(init, 'contract');
+  const outputDir = path.join(init, 'output');
+  mkdirSync(contractDir);
+  mkdirSync(outputDir);
+
+  execSync(`${process.env.FORC_PATH} init`, { cwd: contractDir });
+  execSync(`pnpm fuels init -o ${outputDir} -c ${contractDir} --fuel-core-port 0`, { cwd: init });
+
+  return {
+    init,
+    [Symbol.dispose]: () => {
+      rmSync(init, { recursive: true });
+    },
+  };
+}
+
+/**
+ * @group node
+ */
+describe('dev', () => {
+  it(
+    'cleans up resources on graceful shutdown',
+    async () => {
+      using paths = runInit();
```

Comment:
The `using` keyword is great because you don't have to think about where the test might fail so that you run the `cleanup` if it does fail there. With `using`, the cleanup will happen automatically both when the test finishes as well as when it throws an error.